File: backend/src/main/java/org/jdk/project/service/digitaltwin/DigitalTwinSupport.java
package org.jdk.project.service.digitaltwin;

final class DigitalTwinSupport {

  private DigitalTwinSupport() {}

  static String trimToNull(String value) {
    if (value == null || value.trim().isEmpty()) {
      return null;
    }
    return value.trim();
  }

  static String firstNonBlank(String... values) {
    for (String value : values) {
      String normalized = trimToNull(value);
      if (normalized != null) {
        return normalized;
      }
    }
    return null;
  }

  static Long parseLong(String value) {
    String normalized = trimToNull(value);
    if (normalized == null) {
      return null;
    }
    try {
      return Long.valueOf(normalized);
    } catch (NumberFormatException ignored) {
      return null;
    }
  }
}

File: backend/src/main/kotlin/org/jdk/project/service/digitaltwin/DigitalTwinSupport.kt
package org.jdk.project.service.digitaltwin

object DigitalTwinSupport {
    @JvmStatic
    fun trimToNull(value: String?): String? = value?.trim()?.takeIf { it.isNotEmpty() }

    @JvmStatic
    fun firstNonBlank(vararg values: String?): String? = values.firstNotNullOfOrNull { trimToNull(it) }

    @JvmStatic
    fun parseLong(value: String?): Long? = trimToNull(value)?.toLongOrNull()
}
